src/jit/codegencommon.cpp

```diff
@@ -7329,12 +7335,35 @@ void CodeGen::genProfilingEnterCallback(regNumber initReg, bool* pInitRegZeroed)
                 continue;
             }
 
-            var_types storeType = varDsc->lvaArgType();
-            regNumber argReg    = varDsc->lvArgReg;
-            getEmitter()->emitIns_S_R(ins_Store(storeType), emitTypeSize(storeType), argReg, varNum, 0);
+#ifdef FEATURE_UNIX_AMD64_STRUCT_PASSING
+            if (varTypeIsStruct(varDsc))
+            {
+                var_types       type0, type1;
+                unsigned __int8 offset0, offset1;
+                compiler->GetStructTypeOffset(varDsc->lvVerTypeInfo.GetClassHandle(), &type0, &type1, &offset0,
+                                              &offset1);
+                getEmitter()->emitIns_S_R(ins_Store(type0), emitTypeSize(type0), varDsc->lvArgReg, varNum, offset0);
+                if (type1 != TYP_UNKNOWN)
+                {
+                    getEmitter()->emitIns_S_R(ins_Store(type1), emitTypeSize(type1), varDsc->lvOtherArgReg, varNum,
+                                              offset1);
+                }
+            }
+            else
+#endif
+            {
+                var_types storeType = varDsc->lvaArgType();
+                regNumber argReg    = varDsc->lvArgReg;
+                getEmitter()->emitIns_S_R(ins_Store(storeType), emitTypeSize(storeType), argReg, varNum, 0);
```

Comment:
@sergign60 - as per my understanding the caller of a method doesn't need to ensure stack homing locations for reg args on Unix amd64.  In such a case when the callee method is being jit compiled, this method is trying to home them on to stack.   Either caller or callee method have to ensure stack homing locations for reg args.  Can you please point to me the jit logic ensuring that  when the profiler callback is to be generated?